pkgs/build-support/node/build-npm-package/hooks/npm-build-hook.sh

```diff
@@ -0,0 +1,37 @@
+# shellcheck shell=bash
+
+npmBuildHook() {
+    echo "Executing npmBuildHook"
+
+    if [ -z "${npmBuildScript-}" ]; then
+        echo
+        echo "ERROR: no build script was specified"
+        echo 'Hint: set `npmBuildScript`, override `buildPhase`, or set `dontNpmBuild = true`.'
+        echo
+
+        exit 1
+    fi
+
+    runHook preBuild
+
+    if ! eval "npm run $npmBuildScript $npmBuildFlagsArray $npmFlagsArray"; then
```

Comment:
See https://github.com/NixOS/nixpkgs/pull/189539#discussion_r990434397, here you'll need to `eval "npmFlags=($npmFlags)"` beforehand to make the list propagation work correctly. Same for the other flags, and same for the other build hooks.